DOCS-1052: Add logging steps for MR (#2026)
viamrobotics/docs: docs/modular-resources/create/_index.md

    "go.viam.com/rdk/components/motor"
    "go.viam.com/rdk/resource"
    "go.viam.com/rdk/spatialmath"

>     "go.viam.com/utils"

Reviewer: ```suggestion
```

Unless you are *directly using* a import in go, it will fail to compile.

We already import `golog` so its okay

You need to ensure any dependencies for your module (including the Python or Go [Viam SDK](/program/)) are installed as well.
Your executable will be run by `viam-server` as root, so dependencies need to be available to the root user.

### Configure logging

To enable your module to write logs to the [Viam app](https://app.viam.com/), you must add the following configuration to your respective module code.
Once configured in this way, log messages are sent to the Viam app and appear under the **Logs** tab for your smart machine.

{{< tabs name="Configure logging">}}
{{% tab name="Python"%}}

To enable your Python module to write log messages to the Viam app, add the following lines to your code:

```python {class="line-numbers linkable-line-numbers" data-line="2,5"}
# In your import block, import viam.logging getLogger:
from viam.logging import getLogger

# Before your first class or function, define the LOGGER variable:
LOGGER = getLogger(__name__)
```

{{% /tab %}}
{{% tab name="Go"%}}

To enable your Go module to write log messages to the Viam app, add the following lines to your code:

```go {class="line-numbers linkable-line-numbers"}

> // In your import() block, import the utils package from Viam:

Reviewer: ```suggestion
// In your import() block, import the golog package:
```

import(
       ...
       "github.com/edaniels/golog"
)
// Alter your component to hold a logger
type component struct {
    ...
 logger golog.Logger
}
// Then, alter your component's constructor to save the logger:
func init() {
 registration := resource.Registration[resource.Resource, *Config]{
  Constructor: func(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger golog.Logger) (resource.Resource, error) {
     ...

>      &component {
>          ...
>          logger: logger
>      }

Reviewer: ```suggestion
     return &component {
         ...
         logger: logger
     }, nil
```

Confused rust vs go syntax. My bad.

You need to ensure any dependencies for your module (including the Python or Go [Viam SDK](/program/)) are installed as well.
Your executable will be run by `viam-server` as root, so dependencies need to be available to the root user.

### Configure logging

To enable your module to write logs to the [Viam app](https://app.viam.com/), you must add the following configuration to your respective module code.
Once configured in this way, log messages are sent to the Viam app and appear under the **Logs** tab for your smart machine.

{{< tabs name="Configure logging">}}
{{% tab name="Python"%}}

To enable your Python module to write log messages to the Viam app, add the following lines to your code:

> ```python {class="line-numbers linkable-line-numbers"}

Reviewer 2: ```suggestion
```python {class="line-numbers linkable-line-numbers" data-line="2,5"}
```

# In your import block, import viam.logging getLogger:
from viam.logging import getLogger

# Before your first class or function, define the LOGGER variable:
LOGGER = getLogger(__name__)
```

{{% /tab %}}
{{% tab name="Go"%}}

To enable your Go module to write log messages to the Viam app, add the following lines to your code:

> ```go {class="line-numbers linkable-line-numbers"}

Reviewer 2: ```suggestion
```go {class="line-numbers linkable-line-numbers" data-line="2-5,9,11,13"}
```

> // In your import() block, import the utils package from Viam:
> import(
>        ...
>        "go.viam.com/utils"
> )
> 
> // Then, use the following in your functions to generate log messages as needed:
> // Log with severity warning:
> b.logger.Warnf("error message", b.Name(), err.Error())
> // Log with severity info:
> b.logger.Infof("error message", b.Name(), err.Error())
> // Log with severity debug:
> b.logger.Debugf("error message", b.Name(), err.Error())

Reviewer: ```suggestion
// In your import() block, import the utils package from Viam:
import(
       ...
       "go.viam.com/utils"
)

// Alter your component to hold a logger
type component struct {
    ...
	logger golog.Logger
}

// Then, alter your component's constructor to save the logger
func init() {
	registration := resource.Registration[resource.Resource, *Config]{
		Constructor: func(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger golog.Logger) (resource.Resource, error) {
		   ...
		   &component {
		       ...
		       logger: logger
		   }
		},
	}
	resource.RegisterComponent(...)
}

// Finally, when you need to log, use the functions on your component's logger:
fn (c *component) someFunction(a int) {
  // Log with severity info:
  c.logger.Infof("performing some function with a=%v",a)
  // Log with severity debug (using value wrapping):
  c.logger.Debugw("performing some function","a",a)
  // Log with severity error without a parameter:
  c.logger.Errorln("performing some function")
}

```

Your executable will be run by `viam-server` as root, so dependencies need to be available to the root user.

### Configure logging

To enable your module to write logs to the [Viam app](https://app.viam.com/), you must add the following configuration to your respective module code.
Once configured in this way, log messages are sent to the Viam app and appear under the **Logs** tab for your smart machine.

{{< tabs name="Configure logging">}}
{{% tab name="Python"%}}

To enable your Python module to write log messages to the Viam app, add the following lines to your code:

```python {class="line-numbers linkable-line-numbers" data-line="2,5"}
# In your import block, import viam.logging getLogger:
from viam.logging import getLogger

# Before your first class or function, define the LOGGER variable:
LOGGER = getLogger(__name__)
```

{{% /tab %}}
{{% tab name="Go"%}}

To enable your Go module to write log messages to the Viam app, add the following lines to your code:

```go {class="line-numbers linkable-line-numbers"}
// In your import() block, import the utils package from Viam:
import(
       ...

>        "go.viam.com/utils"

Reviewer: :facepalm: This should be `"github.com/edaniels/golog"`

Missed that.
